### PR TITLE
openai[patch],mistral[patch]: Fix typing for withStructuredOutput

### PR DESCRIPTION
**Pull request by jacoblee93**
@bracesproul

### CODE REVIEW

File: libs/langchain-mistralai/src/chat_models.ts [hunk 1]
```diff
@@ -531,79 +531,45 @@ export class ChatMistralAI<
   }
 
   withStructuredOutput<
-    RunInput = BaseLanguageModelInput,
-    // prettier-ignore
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    RunOutput extends z.ZodObject<any, any, any, any> = z.ZodObject<any, any, any, any>
+    RunOutput extends Record<string, any> = Record<string, any>
```
Comment (bracesproul) [inline libs/langchain-mistralai/src/chat_models.ts:535]: will this still allow for zod schemas to be passed in?

File: libs/langchain-openai/src/chat_models.ts [hunk 2]
```diff
@@ -858,78 +858,45 @@ export class ChatOpenAI<
   }
 
   withStructuredOutput<
-    RunInput = BaseLanguageModelInput,
-    // prettier-ignore
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    RunOutput extends z.ZodObject<any, any, any, any> = z.ZodObject<any, any, any, any>
+    RunOutput extends Record<string, any> = Record<string, any>
```
Comment (bracesproul) [inline libs/langchain-openai/src/chat_models.ts:862]: same here

**Review by bracesproul — APPROVED** [review]

File: libs/langchain-mistralai/src/chat_models.ts (hunk 1 above)
Comment (jacoblee93) [reply libs/langchain-mistralai/src/chat_models.ts:535]: No

File: libs/langchain-openai/src/chat_models.ts (hunk 2 above)
Comment (jacoblee93) [reply libs/langchain-openai/src/chat_models.ts:862]: No

**Pull request merged**